protocol/v2/qbft/storage/ibft_store.go

```diff
@@ -1,29 +1,40 @@
 package qbftstorage
 
 import (
+	"encoding/json"
+
 	specqbft "github.com/bloxapp/ssv-spec/qbft"
 	spectypes "github.com/bloxapp/ssv-spec/types"
 )
 
-// DecidedMsgStore manages persistence of messages
-type DecidedMsgStore interface {
-	GetHighestDecided(identifier []byte) (*specqbft.SignedMessage, error)
-	// SaveHighestDecided saves the given decided message, after checking that it is indeed the highest
-	SaveHighestDecided(signedMsg ...*specqbft.SignedMessage) error
-	// GetDecided returns historical decided messages in the given range
-	GetDecided(identifier []byte, from specqbft.Height, to specqbft.Height) ([]*specqbft.SignedMessage, error)
-	// SaveDecided saves historical decided messages
-	SaveDecided(signedMsg ...*specqbft.SignedMessage) error
-	// CleanAllDecided removes all decided & last decided for msgId
-	CleanAllDecided(msgID []byte) error
+// StoredInstance contains instance state alongside with a decided message (aggregated commits).
+type StoredInstance struct {
+	State          *specqbft.State
+	DecidedMessage *specqbft.SignedMessage
+}
+
+// Encode returns a StoredInstance encoded bytes or error.
+func (si *StoredInstance) Encode() ([]byte, error) {
+	return json.Marshal(si)
+}
+
+// Decode returns error if decoding failed.
+func (si *StoredInstance) Decode(data []byte) error {
+	return json.Unmarshal(data, &si)
 }
 
 // InstanceStore manages instance data
 type InstanceStore interface {
 	// SaveHighestInstance saves the state for the highest instance
-	SaveHighestInstance(state *specqbft.State) error
+	SaveHighestInstance(instance *StoredInstance) error
 	// GetHighestInstance returns the state for the highest instance
-	GetHighestInstance(identifier []byte) (*specqbft.State, error)
+	GetHighestInstance(identifier []byte) (*StoredInstance, error)
+	// GetInstance returns historical decided messages in the given range
+	GetInstance(identifier []byte, from specqbft.Height, to specqbft.Height) ([]*StoredInstance, error)
```

Comment:
rename to `GetInstances` or `GetInstanceInRange`

Reply:
Is `GetInstanceInRange` ok? Renamed